sympy/stats/stochastic_process_types.py

```diff
@@ -971,37 +950,372 @@ def probability(self, condition, given_condition=None, evaluate=True, **kwargs):
+
+    def split(self, l1, l2):
+        if _sympify(l1 + l2) != self.lamda:
+            raise ValueError("Sum of l1 and l2 should be %s" % str(self.lamda))
+        return PoissonProcess(Dummy("l1"), l1), PoissonProcess(Dummy("l2"), l2)
+
+    def expectation(self, expr, condition=None, evaluate=True, **kwargs):
+        r"""
+        Computes expectation
+
+        Parameters
+        ==========
+
+        expr: RandomIndexedSymbol, Relational, Logic
+            Condition for which expectation has to be computed. Must
+            contain a RandomIndexedSymbol of the process.
+        condition: Relational, Boolean
+            The given conditions under which computations should be done, i.e,
+            the intervals on which each variable time stamp in expr is defined
+
+        Returns
+        =======
+
+        Expectation of the given expr
+
+        """
+        if condition is not None:
+            intervals, rv_swap = get_timerv_swaps(expr, condition)
+            if len(intervals) == 1 or all(Intersection(*intv_comb) == EmptySet
+                for intv_comb in itertools.combinations(intervals, 2)): # they are independent
+                if expr.is_Add:
+                    return Add(*[self.expectation(arg, condition) for arg in expr.args])
+                expr = expr.subs(rv_swap)
+            else:
+                return Expectation(expr, condition)
+
+        return _expectation_stoch('Poisson', expr, evaluate=evaluate, **kwargs)
+
+    def probability(self, condition, given_condition=None, evaluate=True, **kwargs):
+        r"""
+        Computes probability
+
+        Parameters
+        ==========
+
+        condition: Relational
+            Condition for which probability has to be computed. Must
+            contain a RandomIndexedSymbol of the process.
+        given_condition: Relational, Boolean
+            The given conditions under which computations should be done, i.e,
+            the intervals on which each variable time stamp in expr is defined
+
+        Returns
+        =======
+
+        Probability of the condition
+
+        """
+        if isinstance(condition, Not):
+            return S.One - self.probability(condition.args[0], given_condition, evaluate, **kwargs)
+        if given_condition is not None:
+            intervals, rv_swap = get_timerv_swaps(condition, given_condition)
+            if len(intervals) == 1 or all(Intersection(*intv_comb) == EmptySet
+                for intv_comb in itertools.combinations(intervals, 2)): # they are independent
+                if isinstance(condition, And):
+                    return Mul(*[self.probability(arg, given_condition) for arg in condition.args])
+                elif isinstance(condition, Or):
+                    return Add(*[self.probability(arg, given_condition) for arg in condition.args])
+                condition = condition.subs(rv_swap)
+            else:
+                return Probability(condition, given_condition)
```

Comment:
Is a parser really needed if `_rv_indexed_subs` is replacing all the RIS with RVs?

Reply:
Yes, because, it will then consider all rvs are independent and produce wrong results